### PR TITLE
Fixes to visualizations throughout the book

### LINKED ISSUE #57: Better explanation of alt.X and alt.Y
Take from 531 lecture notes.

### LINKED ISSUE #128: Simplify plots
For after our initial polish round: the MDS students tried to replicate the plots of the R version (down to fairly detailed stylistic stuff), but we are probably better off just sticking with Altair defaults for the most part as long as they're reasonable. This will keep the plotting code simple and unobtrusive.

### PR DESCRIPTION
**Pull request by joelostblom**
@lheagy @trevorcampbell I had some free time and started fixing some of the open viz issues we have and noticed a few things along the way, so I did a full review on the viz sections of the entire book. Although this PR includes quite a lot of changes, most are clarifying in nature, either making the explanations more intuitive or slightly changing how we create a viz. I also removed any leftovers from the ggplot translation to make it easier for students when they see altair code online such as using the documentation. Lastly I standardized the syntax in all chapters to what was used in the viz chapter so that we don't mix different types of syntax throughout the book. The commit messages include more details about the changes.

I had a quick look at the worksheets and I think the only thing that would need updating there is the scaffold code that students use to match with the syntax in the viz chapter. I can have a more throughout review of the worksheets after you have reviewed this PR, but no hurry.

close #57 close #128

### CODE REVIEW

**Comment by trevorcampbell** [comment]
Great stuff @joelostblom -- I imagine this is one for the end of semester (since the course is in full swing at this point)

**Comment by joelostblom** [comment]
Yeah probably, at least not merging it to production before the semester is over for sure.

File: source/classification2.md
Comment (trevorcampbell) [inline source/classification2.md:330]: The axes on Figure 6.3 look off to me -- it's very squished horizontally. Seems like perhaps the plot is trying to force include the origin for some reason?

Comment (trevorcampbell) [reply source/classification2.md:330]: (I dont think this is related to a change you made though, but something we should fix in this PR)

File: source/regression2.md
Comment (trevorcampbell) [inline source/regression2.md:277]: We should be as consistent as possible everywhere -- do we use `(x="x", y="y")` everywhere, or `alt.X("x")` etc everywhere? should pick one or the other and stick with it, ideally, assuming that's possible

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:88]: the original text here is grammatically correct -- should be "types of plot" not "types of plots"

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:282]: is "altair" typeset using code or text? `altair` vs altair. We should be consistent throughout. I believe in the R book, packages are typeset using code syntax, so here we would prefer `altair`, if I recall

Comment (trevorcampbell) [reply source/viz.md:282]: (this will inevitably be a formatting pass that we have to do later on though)

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:417]: is this an error we made in the R book too?

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:507]: I gotta say: it's totally insane to me that altair treats "point" as "not-filled-in circle", and "circle" as "filled disk". I think the vast majority of people would consider a "point" to be a "dot" or "small filled circle", and "circle" to mean "not-filled-in circle". Blech

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:617]: "in the previous section," <-- add comma

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:618]: awkward switch from "you" to "we"

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:782]: Altair vs `altair` vs altair like above

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:815]: "log adjusted" -> "log-adjusted"

"are shown and the" -> "are shown. The"  <--- separate into two sentences

"indicate" -> "indicates"  typo

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:855]: this is a nice way to format a number, but have we explained to students that we can do this? This should probably appear in an earlier chapter if not...or maybe here if it's the first time it's relevant / used

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: separate long parenthetical out into its own sentence, also edit the new sentence to shorten it a bit for clarity

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: remove comma

Comment (trevorcampbell) [reply source/viz.md]: also "in the altair documentation" <-- specify altair

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: missing period after figure reference

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:1106]: this is cool, but we need to be very careful with how we include this due to the print version of the book obviously not being able to show interactive figures. 

Is it possible to render a different image in the PDF version of the book that is just a static screenshot of what the tooltip would look like? If not immediately, please open an issue to sort that out (and anywhere else we add interactive figures where it's important that the interactivity is displayed -- though I think it will only matter here in Ch4)

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md:1145]: stuff like "hover over things with the mouse" do not look great in a printed / PDF book

Comment (trevorcampbell) [reply source/viz.md:1145]: (but it's good for the web version, hence trying to render two separate copies)

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: oxford comma pls

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: I think this nuance is too much for this relatively early chapter and will confuse a lot of our students. I suggest removing it. 

It's a good point though. If we expanded our chapter with boxplots, I think that's where you'd make this argument w/o losing people. perhaps open an issue to add a section on boxplots, copy/paste this text into that issue so we can save it for later, and delete the note box here.

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: "landmass" isn't a word, it's a variable name -- I would just say "labels" rather than "landmass labels" anyway

Comment (trevorcampbell) [reply source/viz.md]: apparently it is a "newer spelling" and "land mass" is the more traditional. i guess it doesn't matter much, maybe we can just keep landmass

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: this isn't the question -- the question should be 

"Are the top 7 largest landmasses continents?" 

which is why you need "in the affirmative" after.

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: remove "or not"

Comment (trevorcampbell) [reply source/viz.md]: "they are" -> "they correspond to"

Comment (trevorcampbell) [reply source/viz.md]: here is what i'm changing the R version to read:

image

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: remove comma "If instead"

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: NB: this stuff about titles seems to have been removed from the R version. I put a pretty high value on syncing the two books as much as possible to avoid divergent material, so we should decide whether this is included or removed or de-synced from R

Comment (trevorcampbell) [reply source/viz.md]: (though obviously there will be differences due to differing syntax / code patterns etc, but the set of "ideas/topics we teach" should flow the same)

**Review by trevorcampbell — CHANGES_REQUESTED** [review]
Amazing improvement @joelostblom ! I have lots of fiddly comments but overall it's a great update.

**Comment by trevorcampbell** [comment]
> I can have a more throughout review of the worksheets after you have reviewed this PR, but no hurry.

Oh also yeah it would be awesome if you could open a PR that syncs the worksheets/tutorials with your new material too :)

(or at least open an issue that specifies what would need to change)

File: source/viz.md
Comment (joelostblom) [reply source/viz.md:507]: 100% agree, I'm used to it now, but it was definitely one of the things I reacted to as unintuitive when starting to use the library.

File: source/classification2.md
Comment (joelostblom) [reply source/classification2.md:330]: Changed from

image


to

image

File: source/regression2.md
Comment (joelostblom) [reply source/regression2.md:277]: I believe we only use the `alt.X/Y` syntax when setting an option. Although technically we could always use it, even when not setting options, it is often convenient to write just `x='x'` instead of `alt.X('x')`, but I'm not strongly against changing to always use `alt.X/Y` since that would allow us to not specify the named parameter, however I don't think it is an important change and code in the wild + the docs  often uses both syntax so it is good to know of them.

File: source/viz.md
Comment (joelostblom) [reply source/viz.md:88]: Hmm, wouldn't this be similar to saying something like "The types of friends I’ve made in this college are great." where both are plural? Or the "The types of apples that I like are red." Both from this article https://grammarhow.com/type-of-types-of-followed-by-singular-or-plural/. This other article suggest that it might be a difference between American and British English https://english.stackexchange.com/a/5546, with the "type of plurals" being AmE which is what we use in the rest of the book. (I'm realizing I'm ESL and might be missing something but got interested and had to look it up =p )

File: source/viz.md
Comment (joelostblom) [reply source/viz.md:282]: In the docs, we just call it "Altair" without typesetting as code and with a leading capital. I agree we should change it in a later formatting pass if at all.

File: source/viz.md
Comment (joelostblom) [reply source/viz.md:417]: No, it is fine in the R book, but Altair will switch to include the name of the months if we only do 90-94 which is why I extended it with one year

image

versus

image

File: source/viz.md
Comment (joelostblom) [reply source/viz.md:855]: Added a one sentence explanation

File: source/viz.md
Comment (joelostblom) [reply source/viz.md:1106]: Good point, tracking in https://github.com/UBC-DSCI/introduction-to-datascience-python/issues/175

File: source/viz.md
Comment (joelostblom) [reply source/viz.md:1145]: Also in https://github.com/UBC-DSCI/introduction-to-datascience-python/issues/175

File: source/viz.md
Comment (joelostblom) [reply source/viz.md]: Hmm this is what we have in the question further up:

image

Which to me is more similar to "Which of the top 7 largest landmasses are continents?" (fixing a couple of typos)

File: source/viz.md
Comment (joelostblom) [reply source/viz.md]: Updated based on your screenshot with a slight tweak in the wording to match altair's syntax

File: source/viz.md
Comment (joelostblom) [reply source/viz.md]: I agree that it would be valuable to keep them in sync and would prefer to have this section added to the R book. I think it is important to mention titles and show examples of what an appropriate title looks like.

**Review by joelostblom — COMMENTED** [review]
Thanks for the thorough review @trevorcampbell ! I think I have addressed all your comments or opened issues to follow up so please have a look.

File: source/viz.md
Comment (trevorcampbell) [reply source/viz.md:88]: honestly it's so minor that I don't mind either way. 

I don't know the actual rule, and I was just parsing it logically: "types of plot" means multiple kinds of plot. "types of plots" means multiple kinds of...multiple plots?

File: source/viz.md
Comment (trevorcampbell) [reply source/viz.md:417]: good catch!

File: source/viz.md
Comment (trevorcampbell) [reply source/viz.md]: We should keep the wording the same throughout if possible -- so "are the continents the largest 7" or the like

(also if the original Q were just "which of the top 7 are continents" I'd just plot 7 instead of whatever we're doing now, 12 or something like that)

**Comment by trevorcampbell** [comment]
@joelostblom thanks for the fixes. I just have that one remaining comment about the wording of the Q. I think we should keep it as consistent as possible throughout the chapter. Don't see a need to change it from what's in the R version and in this one earlier in the chapter

Aside from that, looks good to me!

File: source/viz.md
Comment (joelostblom) [reply source/viz.md]: I guess it makes it more complicated that our original question is two-fold: both "Are the continents (North / South America, Africa, Europe, Asia, Australia, Antarctica) Earth’s seven largest landmasses?" and then "If so, what are the next few largest landmasses after those?". I updated to include both of those with the same phrasing as in the original question to keep it consistent, let me know what you think.

I believe the reason we have 12 instead of 7 is to show of the color encoding which would just contain one color if we only included the top 7 since those are all continents:

image

File: source/viz.md
Comment (joelostblom) [reply source/viz.md]: Sorry I missed this comment initially. I think this is more than a nuance and a common mistake I see students making, so I would be in favor of adding it to the R and Python books. I think it could fit nicely as a motivation for the histogram section, which could begin by comparing the average and then showing the actual histograms of all observations instead. What do you think of that?

File: source/viz.md
Comment (trevorcampbell) [reply source/viz.md:282]: We have a style guide for the R textbook here: https://github.com/ubc-dsci/introduction-to-datascience#style-guide

we can just follow this (and update it as necessary) once we are at the level of "final polish" for the official publication

File: source/viz.md
Comment (trevorcampbell) [reply source/viz.md]: I'm still not entirely convinced...

Idea: can we open a separate PR (or an issue) to discuss this change? I don't want to block this PR on that

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: @joelostblom typo added here "TO" should be "To"

File: source/viz.md
Comment (trevorcampbell) [inline source/viz.md]: @joelostblom typo "set the"

File: source/viz.md
Comment (trevorcampbell) [reply source/viz.md]: @joelostblom it took me a bit to dig around in my memory for why we decided against this in R, but I finally remembered: we don't use titles anywhere else in the book. If we're suggesting Ch4 is the guide to making good viz, we should follow our own rules elsewhere (and we do, generally). But we felt strongly that we shouldn't have titles everywhere else in the book -- at a minimum, the publisher wouldn't want them in the print book format. So since we weren't including titles everywhere, we didn't feel it would be good to suggest it in Ch4.

I was originally more on the fence about this, but now I'm a solid "no titles" to avoid causing a huge transitive-closure headache elsewhere

File: source/viz.md
Comment (joelostblom) [reply source/viz.md]: Opened https://github.com/UBC-DSCI/introduction-to-datascience-python/issues/178

File: source/viz.md
Comment (joelostblom) [reply source/viz.md]: In the book we do have captions everywhere, which fill a similar need as the titles we are explaining to students and help clarify what is figure contains. I think this is important and we don't want student to produce charts without any explanation (whether it is a title and caption), but removing for now so that we can move forward. Follow up in https://github.com/UBC-DSCI/introduction-to-datascience-python/issues/179

**Comment by joelostblom** [comment]
@trevorcampbell Thanks for the comments, I think I have addressed everything now!

**Review by trevorcampbell — APPROVED** [review]
It's excellent! Thanks so much for the effort @joelostblom -- a nice improvement here.

**Pull request merged**
